fala/apps/adviser/tests/test_crown_dependencies_view.py

```diff
@@ -8,46 +8,42 @@ class PostcodeValidationTest(SimpleTestCase):
     client = Client()
     url = reverse("search")
 
-    def test_guernsey_postcode(self):
-        data = {"postcode": "GY1"}
-        response = self.client.get(self.url, data)
-        self.assertContains(response, "The postcode GY1 is in Guernsey")
-
-    def test_invalid_guernsey_postcode(self):
-        data = {"postcode": "GY1"}
-        response = self.client.get(self.url, data)
-        self.assertContains(response, "Legal Aid in Guernsey")
-
-    def test_invalid_scottish_postcode(self):
-        data = {"postcode": "AB"}
-        response = self.client.get(self.url, data)
-        self.assertContains(response, "The postcode AB is in Scotland")
-
-    def test_valid_scottish_postcode(self):
-        data = {"postcode": "AB11 5BN"}
-        response = self.client.get(self.url, data)
-        self.assertContains(response, "The postcode AB11 5BN is in Scotland")
-
-    def dont_test_valid_english_postcode(self):
-        data = {"postcode": "M2 3WQ"}
-        response = self.client.get(self.url, data)
-        self.assertContains(response, "in order of closeness to")
+    def test_region_postcodes(self):
+        test_cases = [
+            {"postcode": "GY1", "message": "The postcode GY1 is in Guernsey"},
+            {"postcode": "BT93 8AD", "message": "The postcode BT93 8AD is in Northern Ireland"},
+            {"postcode": "IM4", "message": "The postcode IM4 is in the Isle of Man"},
+            {"postcode": "JE2 3FN", "message": "The postcode JE2 3FN is in Jersey"},
+            # English Postcode
+            {"postcode": "M2 3WQ", "message": "in order of closeness to"},
+            # Scottish Postcode
+            {"postcode": "AB11 5BN", "message": "These results cover England and Wales."},
+            # Lower case Postcode
+            {"postcode": "im4", "message": "The postcode IM4 is in the Isle of Man"},
+            # Invalid post code
+            {"postcode": "AB", "message": "in order of closeness to"},
+            # Invalid post code
+            {"postcode": "BT", "message": "in order of closeness to"},
+        ]
+
+        for case in test_cases:
+            with self.subTest(postcode=case["postcode"]):
+                data = {"postcode": case["postcode"]}
+                response = self.client.get(self.url, data)
+                self.assertContains(response, case["message"])
 
     def test_change_search_button_and_form_is_visible(self):
```

Comment:
test confused me a little - could we make what the test is doing a little more explicit in the method like:

def test_other_region_form_and_change_search_button_visible

Reply:
I like it, changed.